src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx

```diff
@@ -269,3 +270,151 @@ describe('DataSourceAggregatedView: read active view (displayAllCompatibleDataSo
+  let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
+  let client: SavedObjectsClientContract;
+  const { toasts } = notificationServiceMock.createStartContract();
+  const uiSettings = uiSettingsServiceMock.createStartContract();
+  const application = applicationServiceMock.createStartContract();
+  const nextTick = () => new Promise((res) => process.nextTick(res));
+
+  beforeEach(() => {
+    client = {
+      find: jest.fn().mockResolvedValue([]),
+    } as any;
+    mockResponseForSavedObjectsCalls(client, 'find', {});
+    mockUiSettingsCalls(uiSettings, 'get', 'test1');
+    jest.spyOn(utils, 'getApplication').mockReturnValue(application);
+  });
+
+  it.each([
+    {
+      filter: (ds: SavedObject<DataSourceAttributes>) => {
+        return true;
+      },
+      activeDataSourceIds: undefined,
+      hideLocalCluster: true,
+      displayAllCompatibleDataSources: true,
+    },
+    {
+      filter: (ds: SavedObject<DataSourceAttributes>) => {
+        return false;
+      },
+      activeDataSourceIds: undefined,
+      hideLocalCluster: true,
+      displayAllCompatibleDataSources: true,
+    },
+    {
+      filter: (ds: SavedObject<DataSourceAttributes>) => {
+        return true;
+      },
+      activeDataSourceIds: undefined,
+      hideLocalCluster: true,
+      displayAllCompatibleDataSources: false,
+    },
+    {
+      filter: (ds: SavedObject<DataSourceAttributes>) => {
+        return false;
+      },
+      activeDataSourceIds: undefined,
+      hideLocalCluster: true,
+      displayAllCompatibleDataSources: false,
+    },
+  ])(
+    'should render warning when no data sources added',
+    async ({ filter, activeDataSourceIds, hideLocalCluster, displayAllCompatibleDataSources }) => {
+      component = shallow(
+        <DataSourceAggregatedView
+          fullWidth={false}
+          hideLocalCluster={hideLocalCluster}
+          savedObjectsClient={client}
+          notifications={toasts}
+          displayAllCompatibleDataSources={displayAllCompatibleDataSources}
+          uiSettings={uiSettings}
+          activeDataSourceIds={activeDataSourceIds}
+          dataSourceFilter={filter}
+        />
+      );
+
+      expect(component).toMatchSnapshot();
+      await nextTick();
+      expect(toasts.add).toBeCalled();
+      expect(component.state('showEmptyState')).toBe(true);
+    }
+  );
+});
+
+describe('DataSourceAggregatedView error state test no matter hide local cluster or not', () => {
+  let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
+  let client: SavedObjectsClientContract;
+  const { toasts } = notificationServiceMock.createStartContract();
+  const uiSettings = uiSettingsServiceMock.createStartContract();
+  const application = applicationServiceMock.createStartContract();
+  const nextTick = () => new Promise((res) => process.nextTick(res));
+
+  beforeEach(() => {
+    client = {
+      find: jest.fn().mockResolvedValue([]),
```

Comment:
how did we simulate the error state?

Reply:
use this mock function can mock the error state:
```
    mockErrorResponseForSavedObjectsCalls(client, 'find');

```

Reply:
in this case, do we still need to initialize client object here?